### DISCUSSION TITLE
Add full path for UI buttons.

### DISCUSSION THREAD

**Discussion by Senior-S** [SmartlyDressedGames/Unturned-3.x-Community#4473] [Ideas]
Hi @SDGNelson, recently a update was made to the EffectManager class, allowing plugin developers to specify the full path of an object, making easier to work with UI's, can be this replicated to the buttons and input field events?
Adding the full path to the button/input field will help a lot while working with UI's.

I know that maybe if you modify the actual event some plugins may broke, but maybe just a new event can be added with this feature.

**Comment by SDGNelson** [comment]
I agree it would be useful information to have. That said, as you mention, it would break existing plugins and necessitate a new event. The reason I added the path option was in consideration of the terrible search performance with thousands of game objects and it didn't affect compatibility, whereas providing the full path in events wouldn't really change anything.

I sympathize it's got to be annoying that I haven't added "good enough" improvements to the existing UI features in favor of some hypothetical future plugin UI support. :[
